src/rgw/rgw_op.cc

```diff
@@ -5282,32 +5298,38 @@ void RGWRestoreObj::execute(optional_yield y)
   int op_ret = s->object->get_obj_attrs(y, this);
   if (op_ret < 0) {
     ldpp_dout(this, 1) << "failed to fetch get_obj_attrs op ret = " << op_ret << dendl;
+    restore_op = op_ret;
     return;
   }
-  rgw::sal::Attrs attrs = s->object->get_attrs();
-  auto attr_iter = attrs.find(RGW_ATTR_MANIFEST);
-  if (attr_iter != attrs.end()) {
-    RGWObjManifest m;
-    decode(m, attr_iter->second);
-    RGWObjTier tier_config;
-    m.get_tier_config(&tier_config);
-    if (m.get_tier_type() == "cloud-s3") {
-      ldpp_dout(this, 20) << "execute: expiry days" << expiry_days <<dendl;
-      op_ret = handle_cloudtier_obj(s, this, driver, attrs, false, expiry_days, true, y);
-      if (op_ret < 0) {
-        ldpp_dout(this, 4) << "Cannot get cloud tiered object: " << *s->object
-        <<". Failing with " << op_ret << dendl;
-        if (op_ret == -ERR_INVALID_OBJECT_STATE) {
-          s->err.message = "This object was transitioned to cloud-s3";
-        }
-      }
-    } else {
-      ldpp_dout(this, 20) << "not cloud tier object erroring" << dendl;
-      op_ret = -ERR_INVALID_OBJECT_STATE;
+  attrs = s->object->get_attrs();
+  auto expire_iter = attrs.find(RGW_ATTR_RESTORE_EXPIRY_DATE);
```

Comment:
These are no needed now in execute(). Please move them to response and that too in the checks wherever its needed (like only in case of CloudRestored). Please cleanup all attr related changes in this routine.